### PR TITLE
User post

### PR DESCRIPTION
**Pull request by jeetsdev**
## Added user and post features 

- All posts
- Users data
- New post
- User profile page
- Edit Post
- Delete post
- Like/dislike 
- Bookmark

### CODE REVIEW

File: src/components/Cards/NewPost/NewPostCard.jsx [hunk 1]
```diff
@@ -1,17 +1,38 @@
...
+
+    const { userData, authToken } = useSelector(state => state.auth);
+    const dispatch = useDispatch();
+    const [postData, setPostData] = useState({
+        content: "",
+    })
+
+    const postSubmitHandler = (event) => {
+        event.preventDefault();
+        if (postData.content !== "") {
```
Comment (rahulmistry751) [inline src/components/Cards/NewPost/NewPostCard.jsx:19]: Suggestions: Can also add a condition for not taking space/blank character as an input

File: src/components/Modals/EditPostModal.jsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    const { authToken } = useSelector(state => state.auth);
+    const dispatch = useDispatch();
+    const [postData, setPostData] = useState({
+        content: post?.content,
+    })
+
+    const updatePostHandler = (event) => {
+        console.log('event: ', event);
+        event.preventDefault();
+        if (postData.content !== "") {
```
Comment (rahulmistry751) [inline src/components/Modals/EditPostModal.jsx:21]: Same goes for modal editing

**Review by rahulmistry751 — APPROVED** [review]
Clean UI and code

**Review by Pratik1005 — APPROVED** [review]
Functionality working properly, good to merge

**Pull request merged**